tests/unit/test.program.js

```diff
@@ -430,6 +433,203 @@ describe('program.main', () => {
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          logger,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch(onlyInstancesOf(UsageError, (error) => {
+        assert.match(error.message, /bad path/);
+        sinon.assert.notCalled(fakeApplyConfigToArgv);
+        sinon.assert.notCalled(fakeProcess.exit);
+        sinon.assert.calledOnce(logger.error);
+      }));
+  });
+
+  it('throws when config file loading fails', () => {
+    const fakeProcess = createFakeProcess();
+    const logger = fake(createLogger(__filename));
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      throw new Error('some error');
+    });
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          logger,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch((error) => {
+        if (!(error instanceof UsageError)) {
+          assert.match(error.message, /some error/);
+          sinon.assert.notCalled(fakeApplyConfigToArgv);
+          sinon.assert.notCalled(fakeProcess.exit);
+          sinon.assert.calledOnce(logger.error);
+        }
+      });
+  });
+
+  it('logs and throws when fakeApplyConfigToArgv throws UsageError', () => {
+    const fakeProcess = createFakeProcess();
+    const logger = fake(createLogger(__filename));
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {});
+    const fakeApplyConfigToArgv = sinon.spy(() => {
+      throw new UsageError('bad config option');
+    });
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          logger,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch(onlyInstancesOf(UsageError, (error) => {
+        assert.match(error.message, /bad config option/);
+        sinon.assert.notCalled(fakeProcess.exit);
+        sinon.assert.calledOnce(logger.error);
+      }));
+  });
+
+  it('logs and throws when fakeApplyConfigToArgv throws WebExtError', () => {
```

Comment:
@saintsebastian I'm wondering if we are really supposed to throw a WebExtError when a config file option is missing the type:

https://github.com/mozilla/web-ext/blob/177709b59137b74ca933014e843894b169be6487/src/config.js#L52-L55

while we are raising a UsageError when the specified type is incorrect:

https://github.com/mozilla/web-ext/blob/177709b59137b74ca933014e843894b169be6487/src/config.js#L61-L65

it seems to me that we could/should throw a UsageError for both, @kumar303 what do you think about it?
(e.g. I'm wondering if this has been already evaluated when the config helper module has been created)